mozjs/build.rs

```diff
@@ -131,24 +138,103 @@ fn cc_flags() -> Vec<&'static str> {
     result
 }
 
+fn cargo_target_dir() -> PathBuf {
+    let out_dir = PathBuf::from(env::var_os("OUT_DIR").unwrap());
+    let mut dir = out_dir.as_path();
+    while let Some(target_dir) = dir.parent() {
+        if target_dir.file_name().unwrap().to_string_lossy() == "target" {
+            return target_dir.to_path_buf();
+        }
+        dir = target_dir;
+    }
+    panic!("$OUT_DIR is not in target")
+}
+
+fn find_moztools() -> Option<PathBuf> {
+    let cargo_target_dir = cargo_target_dir();
+    let deps_dir = cargo_target_dir.join("dependencies");
+    let moztools_path = deps_dir.join(MOZ_TOOLS);
+
+    if moztools_path.exists() {
+        Some(moztools_path)
+    } else {
+        None
+    }
+
+    // to support many moztools versions
+    /*
+    if let Ok(dirs) = std::fs::read_dir(deps_dir) {
+        dirs.into_iter().find_map(|p| {
+            if let Ok(pp) = p {
+                if let Some(name) = pp.path().file_name() {
+                    if name.to_string_lossy().contains("moztools") {
```

Comment:
This part is fine if you're still planning to implement it in this PR, but should probably be removed otherwise.